services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MultiRerouteSpec.groovy

```diff
@@ -7,66 +7,79 @@ import static org.openkilda.testing.Constants.WAIT_OFFSET
 import org.openkilda.functionaltests.HealthCheckSpecification
 import org.openkilda.functionaltests.helpers.Wrappers
 import org.openkilda.messaging.info.event.IslChangeType
-import org.openkilda.messaging.payload.flow.FlowPayload
 import org.openkilda.messaging.payload.flow.FlowState
+import org.openkilda.northbound.dto.v2.flows.FlowRequestV2
 
 import java.util.concurrent.TimeUnit
 
 class MultiRerouteSpec extends HealthCheckSpecification {
 
     def "Simultaneous reroute of multiple flows should not oversubscribe any ISLs"() {
         given: "Two flows on the same path, with alt paths available"
-        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find { it.paths.size() > 1 } ?:
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find { it.paths.size() > 2 } ?:
                 assumeTrue("No suiting switches found", false)
-        List<FlowPayload> flows = []
-        2.times {
-            def flow = flowHelper.randomFlow(switchPair)
+        List<FlowRequestV2> flows = []
+        20.times {
+            def flow = flowHelperV2.randomFlow(switchPair)
             flow.maximumBandwidth = 10000
-            flowHelper.addFlow(flow)
+            flowHelperV2.addFlow(flow)
             flows << flow
         }
-        def currentPath = pathHelper.convert(northbound.getFlowPath(flows[0].id))
-        //ensure both flows are on the same path
-        assert pathHelper.convert(northbound.getFlowPath(flows[1].id)) == currentPath
+        def currentPath = pathHelper.convert(northbound.getFlowPath(flows[0].flowId))
+        //ensure all flows are on the same path
+        flows[1..-1].each {
+            assert pathHelper.convert(northbound.getFlowPath(it.flowId)) == currentPath
+        }
```

Comment:
should we check that 20 flows were created?

Reply:
Why wouldn't they? FlowHelper kind of guarantees that all flows are created, why would we recheck this here? Didn't get your idea

Reply:
>  FlowHelper kind of guarantees that all flows are created

agree